jaxley/optimize/transforms.py

```diff
@@ -1,31 +1,153 @@
 # This file is part of Jaxley, a differentiable neuroscience simulator. Jaxley is
 # licensed under the Apache License Version 2.0, see <https://www.apache.org/licenses/>
 
-from typing import Dict, List
+from abc import ABC, abstractmethod
+from typing import Any, Callable, Dict, List, Sequence
 
 import jax.numpy as jnp
+from jax import Array
+from jax.typing import ArrayLike
 
 from jaxley.solver_gate import save_exp
 
 
-def sigmoid(x: jnp.ndarray) -> jnp.ndarray:
-    """Sigmoid."""
-    return 1 / (1 + save_exp(-x))
+class Transform(ABC):
```

Comment:
can it have a `def forward`? And the forward `__call__()` just runs `forward()`?

Reply:
Sure. Actually then also ParamsTransform can subclass it?